vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinWebSecurityConfigurerAdapter.java

```diff
@@ -245,10 +247,63 @@ protected void setLoginView(HttpSecurity http,
         FormLoginConfigurer<HttpSecurity> formLogin = http.formLogin();
         formLogin.loginPage(loginPath).permitAll();
         formLogin.successHandler(
-                new VaadinSavedRequestAwareAuthenticationSuccessHandler());
+                getVaadinSavedRequestAwareAuthenticationSuccessHandler(http));
         http.csrf().ignoringAntMatchers(loginPath);
         http.logout().logoutSuccessUrl(logoutUrl);
         viewAccessChecker.setLoginView(flowLoginView);
     }
 
+    /**
+     * Sets up stateless JWT authentication using cookies.
+     *
+     * @param http
+     *            the http security from {@link #configure(HttpSecurity)}
+     * @param secretKey
+     *            the secret key for encoding and decoding JWTs, must use a
+     *            {@link MacAlgorithm} algorithm name
+     * @param issuer
+     *            the issuer JWT claim
+     * @throws Exception
+     *             if something goes wrong
+     */
+    protected void setJwtSplitCookieAuthentication(HttpSecurity http,
```

Comment:
Is the API name a bit too technical? would it be better with sth like `setStatelessAuthentication()`? so that we also don't bind the API to the implementation details.